src/coreclr/src/pal/src/map/map.cpp

```diff
@@ -2137,6 +2137,18 @@ MAPRecordMapping(
     return palError;
 }
 
+static size_t OffsetWithinPage(off_t addr)
+{
+    return addr & (GetVirtualPageSize() - 1);
+}
+
+static size_t RoundToPage(size_t size, off_t offset)
+{
+    size_t result = size + OffsetWithinPage(offset);
```

Comment:
Wouldn't this be `result = size + (GetVirtualPageSize() - OffsetWithinPage(offset))`?

Reply:
Perhaps naming is not the most intuitive, I did not want to change since I could not come up with something much better. However the math is correct, I think.
The idea is - if the header offset is 4K + 128bytes, we will mmap a larger region starting from 4K, to make mmap happy. The size of the mapping will be expanded by 128, which is `OffsetWithinPage`

Reply:
In that case it should be called something like `GetSizeFromPageBoundary` - or maybe even more specific `GetSizeFromPageStart`...

Or just move the math to the code below - it seems this is called exactly once.

Reply:
`GetSizeFromPageStart` sounds better. It is probably better to do such changes in master. I will log an issue.